partingpets/partingpets_fe
# Mc edit user

## Massive Update
1. Updated Routes in app.js to allow props to be passed.
1. Added Edit User Form functionality on Profile Page
1. Updated .eslintrc to allow ++ in for loops
1. Add update user request
1. Added utility function to generate Partner Registration Codes
1. Update Icons in Profile dropdown to match navbar


# How to Test
1. Go to profile page, Click Edit.  Change Partner to NO. CLick Submit
    - You should lose access to Partner Link on Navbar
    - In USER table in SQL, PartnerId should be set to 0(False) and PartnerID set to Null
1. Go to Profile and click Edit. Change Partner to Yes and Enter `gd5v34rb` for Partner Id. CHange focus out of Registration code.  Form should validate the code.
    - Click submit and you should have access to Partner Navbar item
    - Values in DB should be changed back
1. Go to profile and click Edit.  Change any/all values and click submit.  Values should be saved and the portal updated with new information
